flink-connector-aws/flink-connector-dynamodb/src/main/java/org/apache/flink/connector/dynamodb/source/reader/PollingDynamoDbStreamsShardSplitReader.java

```diff
@@ -75,11 +82,17 @@ public RecordsWithSplitIds<Record> fetch() throws IOException {
             return INCOMPLETE_SHARD_EMPTY_RECORDS;
         }
 
-        GetRecordsResponse getRecordsResponse =
-                dynamodbStreams.getRecords(
-                        splitState.getStreamArn(),
-                        splitState.getShardId(),
-                        splitState.getNextStartingPosition());
+        GetRecordsResponse getRecordsResponse;
+        try {
+            getRecordsResponse =
+                    dynamodbStreams.getRecords(
+                            splitState.getStreamArn(),
+                            splitState.getShardId(),
+                            splitState.getNextStartingPosition());
+        } catch (ResourceNotFoundException e) {
+            return new DynamoDbStreamRecordsWithSplitIds(
+                    Collections.emptyIterator(), splitState.getSplitId(), true);
+        }
```

Comment:
nit: I think this is not needed anymore, because we handle it inside the streamProxy.  No harm keeping though. We should add a WARN log

Reply:
i don't think getRecords in streamProxy handles this.